cmd/zpool/zpool_main.c

```diff
@@ -8888,19 +8888,12 @@ upgrade_cb(zpool_handle_t *zhp, void *arg)
 	}
 
 	if (modified_pool) {
-#ifdef __FreeBSD__
-		char bootfs[ZPOOL_MAXPROPLEN];
-		if (zpool_get_prop(zhp, ZPOOL_PROP_BOOTFS, bootfs,
-		    sizeof (bootfs), NULL, B_FALSE) == 0 &&
-		    strcmp(bootfs, "-") != 0) {
-			(void) printf(gettext("\nPool '%s' has the bootfs "
-			    "property set, you might need to update\nthe boot "
-			    "code. See gptzfsboot(8) and loader.efi(8) for "
-			    "details.\n"), zpool_get_name(zhp));
-		}
-#else
-		(void) printf("\n");
-#endif
+		char msg[MAXPATHLEN];
```

Comment:
I'm a bit wary of making this use a buffer when we could just be printing directly.

Reply:
I wanted to keep printfs out of os specific files (I'm not familiar with the architecture though, it just felt like controlling output in zpool_main.c makes more sense, especially with the printf/newline situation in there).

Reply:
Then again, I could of course just change it, so it's

```c
if (modified) {
  printf("\n");
  do_the_check_that_outputs_stuff();
}

Reply:
That feels the best to me.